src/App.test.js

```diff
@@ -16,14 +16,29 @@ it('renders without crashing', () => {
   shallow(<App />);
 });
 
-it('chatWindow not rendered when authorizationValues state has no access_token', () => {
-  app.setState({authenticationValues: {}})
+it('does not render chatWindow when authorizationValues state has no access_token', () => {
+  app.setState({authenticationInfo: {}})
   expect(app.find("LoginWithAmazon").getElements().length).toBe(1)
   expect(app.find("ChatWindow").getElements().length).toBe(0)
 });
 
-it('chatWindow rendered when authorizationValues state has access_token', () => {
-  app.setState({authenticationValues: {access_token: "some_access_token", expires_in: 30}});
+it('renders chatWindow when authorizationValues state has access_token', () => {
+  app.setState({authenticationInfo: {access_token: "some_access_token", expires_in: 30}});
   expect(app.find("LoginWithAmazon").getElements().length).toBe(1)
   expect(app.find("ChatWindow").getElements().length).toBe(1)
 });
+
+it('should not change state when authorization response (implicit grant) is not defined', () => {
+  appInstance.populateAuthenticationInfo();
+  expect(appInstance.state).toEqual(originalState);
+});
+
+it('should change state when authorization response (implicit grant) is defined', () => {
+  appInstance.populateAuthenticationInfo({access_token: "some_access_token", expires_in: 30, state: "user_defined_state"});
+  const finalState = {
+    authenticationInfo: {
+      access_token: "some_access_token",
```

Comment:
When original and expected values are to be compared, they should be extracted into a single field in tests.